tests/unit-tests/scene/test_surface_stack.cpp

```diff
@@ -1458,4 +1438,53 @@ TEST_F(SurfaceStack, multiple_surfaces_can_be_swapped)
             SceneElementForStream(stub_buffer_stream2)));
 
     Mock::VerifyAndClearExpectations(&observer);
-}
\ No newline at end of file
+}
+
+TEST_F(SurfaceStack, observer_is_notified_on_lock)
+{
+    auto observer = std::make_shared<MockSessionLockObserver>();
+    EXPECT_CALL(*observer, on_lock()).Times(1);
+    stack.register_interest(observer);
```

Comment:
If we remove the `Executor` parameter from `SurfaceStack()` then we'll need to explicitly register with the immediate executor here (and the subsequent places):
```suggestion
    stack.register_interest(observer, mir::immediate_executor);
```